modules/tamr-vm/main.tf

```
@@ -73,5 +73,16 @@ resource "azurerm_virtual_machine" "tamr-vm" {
     }
   }
 
+  plan {
+    count = contains(keys(var.plan), "name") &&
+            contains(keys(var.plan), "publisher") &&
+            contains(keys(var.plan), "product")
+            ? 1 : 0
+
+    name      = var.plan["name"]
```

Is this ok when plan doesn't have a key name?

Do we need to do something like this?
```suggestion
    name      = contains(keys(var.plan), "name") ? var.plan["name"] : null
```

---

The check on line 77 should prevent this block from getting created if name is not specified.

---

How did you confirm that setting the count to 0 doesn't evaluate these statements?

---

Oh yeah, good point. I was incorrectly treating the block like a resource. Changed it to be a dynamic block.

---

Turns out we're already have variables that capture all the info we need for the plan. Changing to use those.